File: signcall-overlay/frontend/src/types/messages.ts
export type FrameIn = {
  type: "frame";
  session: string;
  user: string;
  ts: number;
  image_jpeg_b64: string;
  style?: "concise" | "detailed";
};

export type CaptionOut = {
  type: "caption";
  session: string;
  user: string;
  ts: number;
  caption: string;
  confidence: number;
  mode: "template" | "llm" | "uncertain";
};

export type CorrectionIn = {
  type: "correction";
  session: string;
  user: string;
  ts: number;
  incorrect_token: string;
  correct_token: string;
};

export type WsIn = CaptionOut;
export type WsOut = FrameIn | CorrectionIn;

File: signcall-overlay/backend/app/schemas/messages.py
from pydantic import BaseModel
from typing import Literal, Optional

class FrameIn(BaseModel):
    type: Literal["frame"]
    session: str
    user: str
    ts: int
    image_jpeg_b64: str
    style: Optional[Literal["concise", "detailed"]] = "concise"

class CaptionOut(BaseModel):
    type: Literal["caption"]
    session: str
    user: str
    ts: int
    caption: str
    confidence: float
    mode: Literal["template", "llm", "uncertain"]

class CorrectionIn(BaseModel):
    type: Literal["correction"]
    session: str
    user: str
    ts: int
    incorrect_token: str
    correct_token: str
